tests/mocks.h

```diff
@@ -74,17 +74,17 @@ class MockBPFtrace : public BPFtrace {
     return { "mock_vmlinux" };
   }
 
-  const struct stat &get_pidns_self_stat() const override
+  const std::optional<struct stat> &get_pidns_self_stat() const override
   {
-    static const struct stat init_pid_namespace = []() {
+    static const std::optional<struct stat> init_pid_namespace = []() {
       struct stat s {};
       s.st_ino = 0xeffffffc; // PROC_PID_INIT_INO
-      return s;
+      return std::optional{ s };
```

Comment:
Should be ok to just return `s` like this, right? Same for below

```suggestion
      return s;
```

Reply:
That didn't compile (i guess the closure return type is inferred from the type of the object in the return statement not the type of the variable the result is assigned ti